sdk/push-fcm/src/main/java/com/klaviyo/pushFcm/Notification.kt

```diff
@@ -0,0 +1,140 @@
+package com.klaviyo.pushFcm
+
+import android.Manifest
+import android.app.Notification
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import androidx.core.app.ActivityCompat
+import androidx.core.app.NotificationChannelCompat
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import com.google.firebase.messaging.RemoteMessage
+import com.klaviyo.core.Registry
+
+class Notification(private val message: RemoteMessage) {
+
+    /**
+     * Constants and extension properties
+     *
+     * NOTE: We always send data-only messages, so all the payload keys are of our choosing.
+     *  For consistency, I've chosen to mostly shadow the FCM notification object keys though.
+     *
+     * See Extensions.kt where accessors are defined as property extensions to [RemoteMessage]
+     */
+    companion object {
+        internal const val CHANNEL_ID_KEY = "channel_id"
+        internal const val CHANNEL_NAME_KEY = "channel_name"
+        internal const val CHANNEL_DESCRIPTION_KEY = "channel_description"
+        internal const val CHANNEL_IMPORTANCE_KEY = "channel_importance"
+        internal const val SMALL_ICON_KEY = "small_icon"
+        internal const val TITLE_KEY = "title"
+        internal const val BODY_KEY = "body"
+        internal const val URL_KEY = "url"
+        internal const val CLICK_ACTION_KEY = "click_action"
+        internal const val SOUND_KEY = "sound"
+        internal const val NOTIFICATION_COUNT_KEY = "notification_count"
+        internal const val NOTIFICATION_PRIORITY = "notification_priority"
+
+        /**
+         * Get an integer ID to associate with a notification or its pending intent
+         * The notification system service will de-dupe on this ID alone,
+         * and I don't think we want our notifications to be de-duped
+         *
+         * NOTE: The FCM SDK also uses a timestamp to construct its integer IDs
+         */
+        private fun generateId() = Registry.clock.currentTimeMillis().toInt()
+    }
+
+    /**
+     * Formats and displays a notification based on the remote message data payload
+     *
+     * @param context
+     */
+    fun displayNotification(context: Context) {
+        if (ActivityCompat.checkSelfPermission(
+                context,
+                Manifest.permission.POST_NOTIFICATIONS
+            ) != PackageManager.PERMISSION_GRANTED
```

Comment:
Curious how adding this in the manifest translates to actually having the right permissions?

<img width="948" alt="image" src="https://user-images.githubusercontent.com/118314354/226748060-b9b5764a-3bce-4057-8f93-ac567142cce8.png">

Reply:
Kind of like having to put that you need a permission into your `plist`, but also still needing to ask the user for that permission at runtime. This is the runtime check

Reply:
Hmm, so if I didn't add this in and prompted the user for permission from the host app, the end user wouldn't see that prompt? Just trying to understand the underlying reason for this thing.

Reply:
Got it. If I'm making the right connection, this is like adding the entitlement in iOS.